### PR TITLE
Path

### PR DESCRIPTION
**Pull request by ericgaspar**

### CODE REVIEW

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

**Comment by silverhook** [comment]
Will this patch fix the issue where BorgWarehouse does not want to create a repo because it is missing `~/.ssh/authorized_keys`?

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
> Will this patch fix the issue where BorgWarehouse does not want to create a repo because it is missing `~/.ssh/authorized_keys`?

This PR needs to be tested

**Comment by silverhook** [comment]
@ericgaspar, can I help in any way?

**Comment by ericgaspar** [comment]
if you are willing to test it will help 👍

**Comment by silverhook** [comment]
If you can walk me through what I should do, happy to.

(I’m new to YunoHost, but not new to Linux)

**Comment by silverhook** [comment]
@ericgaspar , if you release a testing version on YunoHost, I’d be glad to take it for a spin and report my experience.

**Comment by ericgaspar** [comment]
You can test (install or upgrade) this branch with this commands in a terminal
```
sudo yunohost app install https://github.com/YunoHost-Apps/borgwarehouse_ynh/tree/path
or
sudo yunohost app upgrade borgwarehouse -u https://github.com/YunoHost-Apps/borgwarehouse_ynh/tree/path --force
```

**Comment by silverhook** [comment]
Thanks, @ericgaspar ,

I just did the latter (force upgrade) and restarted the service, but when I try to add a repository I still get:

> An error has occurred : /home/borgwarehouse/.ssh/authorized_keys must be present

File: scripts/install [hunk 1]
```diff
@@ -38,11 +38,13 @@ ynh_script_progression --message="Setting up source files..." --weight=1
 # Download, check integrity, uncompress and patch the source from manifest.toml
 ynh_setup_source --dest_dir="$install_dir/app"
 
-mkdir $install_dir/.ssh && chmod 700 $install_dir/.ssh
-touch $install_dir/.ssh/authorized_keys && chmod 600 $install_dir/.ssh/authorized_keys
+mkdir /home/$app
+mkdir /home/$app/.ssh && chmod 700 /home/$app/.ssh
+touch /home/$app/.ssh/authorized_keys && chmod 600 /home/$app/.ssh/authorized_keys
```
Comment (silverhook) [inline scripts/install:43]: Perhaps add these also to `script/upgrade`

**Comment by silverhook** [comment]
OK, I am running into another problem: I can create backup repos, but cannot seem to access them.

If I try to `borg init` a repo from my laptop that I created using BorgWarehouse@YunoHost, I get the following error:

```
borgwarehouse@{$domain}'s password: Remote: Permission denied, please try again.

```

I have no idea which password it wants from me.

I also tried to [import the repository from my old borg server](https://borgwarehouse.com/docs/admin-manual/import-old-repo/), but it does not seem to show up as used space in the dashboard.

That brings me to the question where does BorgWarehouse on YunoHost expect the `repos/` to be?

BTW, I don’t know if this affects anything, but even though there is `/home/borgwarehouse/`, borgwarehouse’s home directory is still set to `/var/www/borgwarehouse/`.

**Comment by silverhook** [comment]
I tried again to init a backup repository with `borg init --debug -e repokey-blake2 […]` and I get:

```
sing builtin fallback logging configuration
33 self tests completed in 0.04 seconds
SSH command line: ['ssh', '-p', '22', '{redacted}', 'borg', 'serve', '--debug']
Remote: ssh: connect to host {redacted} port 22: Connection refused
Connection closed by remote host. Is borg working on the server?
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/borg/archiver.py", line 5343, in main
    exit_code = archiver.run(args)
                ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/borg/archiver.py", line 5263, in run
    return set_ec(func(args))
                  ^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/borg/archiver.py", line 167, in wrapper
    repository = RemoteRepository(location.omit_archive(), create=create, exclusive=argument(args, exclusive),
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/borg/remote.py", line 578, in __init__
    raise ConnectionClosedWithHint('Is borg working on the server?') from None
borg.remote.ConnectionClosedWithHint: Connection closed by remote host. Is borg working on the server?

Platform: Linux leza 6.6.8-zen1-1-zen #1 ZEN SMP PREEMPT_DYNAMIC Thu, 21 Dec 2023 19:00:41 +0000 x86_64
Linux: Unknown Linux  
Borg: 1.2.7  Python: CPython 3.11.6 msgpack: 1.0.5 fuse: llfuse 1.5.0 [pyfuse3,llfuse]
PID: 239375  CWD: /home/hook
sys.argv: ['/usr/bin/borg', 'init', '--debug', '-e', 'repokey-blake2', 'ssh://borgwarehouse@{redacted}']
SSH_ORIGINAL_COMMAND: None
```

**Comment by silverhook** [comment]
@ericgaspar , anything else I can help with?

I’m interested to getting this fixed, as migrating my Borg backups to YuNoHost is the last thing keeping my noisy old server(s) running.

**Comment by silverhook** [comment]
I poked around a bit and it seems to be a permission (and folder location) issue.

**Comment by Limezy** [comment]
@ericgaspar would you mind giving me the rights on this repo ?
I need to use borgwarehouse so I'll solve the issues first

**Comment by Limezy** [comment]
!testme

**Comment by Limezy** [comment]
superseded by #13 after a rebase of these commits

**Pull request merged**